Attempt to infer correct wind units based on Airport prefix

pm-conej opened the pull request:
If the wind speed is missing from the wind token, then determine
	default units for the airport based on airport prefix.  Currently
	this will work for all prefix but 'U'.  If the airport prefix is 'U'
	then leave the units blank.

	Issue: FAPI-902 Incorrect METAR data

pm-conej committed:
Attempt to infer correct wind units based on Airport prefix

	If the wind speed is missing from the wind token, then determine
	default units for the airport based on airport prefix.  Currently
	this will work for all prefix but 'U'.  If the airport prefix is 'U'
	then leave the units blank.

	Issue: FAPI-902 Incorrect METAR data

pm-conej committed:
Include MPS as a valid wind unit

	Issue: FAPI-814 METARS outside the US appear to not be parsed
	correctly.

pm-conej committed:
Fix incorrect pattern matching on missing(slash) fields

	Issue: FAPI-943 Incorrect METAR display

bovine commented on src/decode_metar.c:
```diff
@@ -2223,7 +2231,57 @@ static MDSP_BOOL isTimeUTC( char *UTC, Decoded_METAR *Mptr, int *NDEX )
+   static char *airportPrefixUsingMPS[1] = { "Z" },
+               *airportPrefixWithoutDefault[1] = { "U" };
+
+   char airportPrefix = airportId[0];
```
might be good to also verify that the strlen(airportId) is 4, to guard against non-ICAO airport codes getting passed in somehow.

bovine commented on src/decode_metar.c:
```diff
@@ -2223,7 +2231,57 @@ static MDSP_BOOL isTimeUTC( char *UTC, Decoded_METAR *Mptr, int *NDEX )
+static MDSP_BOOL defaultWindUnitsForAirport( char *airportId, char **defaultUnit )
+{
+   int i;
+   static char *airportPrefixUsingMPS[1] = { "Z" },
```
maybe add a comment saying which geographic areas these prefixes are covering?

bovine commented on src/decode_metar.c:
```diff
@@ -2255,11 +2313,17 @@ static MDSP_BOOL isWindData( char *wind, Decoded_METAR *Mptr, int *NDEX )
    char *GustPtr,
-        *unitsPtr;
+        *unitsPtr,
+        *defaultUnit = malloc(4);
```
Is this malloc() being leaked? I don't see matching free()

pm-conej committed:
Address code review comments

	Add comment to explain airport prefix regions.  Check that airport
	ID is 4 characters long.  Free defaultUnit after it is used.

	Issue: FAPI-902 Incorrect METAR data

pm-conej committed:
Move strlen airportId check below all var declarations

	Issue: FAPI-902 Incorrect METAR data

pm-conej committed:
Change return type for defaultWindUnitsForAirport to char

	Change the return type for defaultWindUnitsForAirport to chat
	and then assign that char to the existing dummy buffer.  If the
	airport prefix does not have a default then just return an empty
	string.

	Issue: FAPI-902 Incorrect METAR data

pm-conej committed:
Return empty string when airportId longer than 4 chars

	Issue: FAPI-902 Incorrect METAR data

pm-conej committed:
Add comment to airportId strlen check for ICOA airport

	Issue: FAPI-902 Incorrect METAR data
